src/index.test.mjs

```diff
@@ -65,64 +65,171 @@ test("astroRehypeRelativeMarkdownLinks", async (t) => {
   );
 
-  await t.test("should keep query and fragment", async () => {
-    const input = '<a href="./fixtures/test.md?q=q#hash">foo</a>';
+  await t.test(
+    "should keep query and fragment for file paths that exist",
+    async () => {
+      const input = '<a href="./fixtures/test.md?q=q#hash">foo</a>';
+      const { value: actual } = await rehype()
+        .use(testSetupRehype)
+        .use(astroRehypeRelativeMarkdownLinks, { contentPath: "src" })
+        .process(input);
+
+      const expected =
+        '<html><head></head><body><a href="/fixtures/test?q=q#hash">foo</a></body></html>';
+
+      assert.equal(actual, expected);
+  });
+
+  await t.test(
+    "should prefix base to output on file paths that exist",
+    async () => {
+      const input = '<a href="./fixtures/test.md">foo</a>';
+      const { value: actual } = await rehype()
+        .use(testSetupRehype)
+        .use(astroRehypeRelativeMarkdownLinks, {
+          contentPath: "src",
+          basePath: "/testBase",
+        })
+        .process(input);
+
+      const expected =
+        '<html><head></head><body><a href="/testBase/fixtures/test">foo</a></body></html>';
+
+      assert.equal(actual, expected);
+  });
+
+  await t.test(
+    "should not replace path if relative file does not exist",
+    async () => {
+      const input = '<a href="./fixtures/does-not-exist.md">foo</a>';
+      const { value: actual } = await rehype()
+        .use(testSetupRehype)
+        .use(astroRehypeRelativeMarkdownLinks, { contentPath: "src" })
+        .process(input);
+
+      const expected =
+        '<html><head></head><body><a href="./fixtures/does-not-exist.md">foo</a></body></html>';
+
+      assert.equal(actual, expected);
+    },
+  );
+
+  await t.test(
+    "should transform file paths that exist with non alphanumeric characters",
+    async () => {
+      const input = '<a href="./fixtures/test%20(non-alpha).md">foo</a>';
+      const { value: actual } = await rehype()
+        .use(testSetupRehype)
+        .use(astroRehypeRelativeMarkdownLinks, { contentPath: "src" })
+        .process(input);
+
+      const expected =
+        '<html><head></head><body><a href="/fixtures/test-non-alpha">foo</a></body></html>';
+
+      assert.equal(actual, expected);
+    },
+  );
+
+  await t.test("should not replace absolute path if file exists", async () => {
+    const absolutePath = path.resolve("./fixtures/test.md");
+    const input = `<a href="${absolutePath}">foo</a>`;
+    const { value: actual } = await rehype()
+      .use(testSetupRehype)
+      .use(astroRehypeRelativeMarkdownLinks, { contentPath: "src" })
+      .process(input);
+
+    const expected = `<html><head></head><body><a href="${absolutePath}">foo</a></body></html>`;
+
+    assert.equal(actual, expected);
+  });
+
+  await t.test(
+    "should not replace absolute path if file does not exist",
+    async () => {
+      const absolutePath = `${path.dirname("./fixtures/test.md")}/does-not-exist.md`;
```

Comment:
I'm a bit confused as this outputs a relative path to test with (`./fixtures/does-not-exist.md`), which makes it like the the existing tests. Is this intended?

Reply:
Sorry, no!  This is a bug in the test, I forgot to resolve to absolute.  I'll update.

Reply:
updated, sorry about that, great catch!